From cf32840c52924fc113a9784743234982253358bb Mon Sep 17 00:00:00 2001
From: Tomas Henzl <thenzl@redhat.com>
Date: Thu, 18 Sep 2014 14:14:02 -0400
Subject: [scsi] megaraid: minor cut and paste error fixed

Message-id: <1411049668-28841-3-git-send-email-thenzl@redhat.com>
Patchwork-id: 96155
O-Subject: [RHEL7.1 e-stor PATCH 02/28] megaraid: minor cut and paste error fixed.
Bugzilla: 1088523
RH-Acked-by: Ewan Milne <emilne@redhat.com>
RH-Acked-by: Rob Evers <revers@redhat.com>

This looks like a cut and paste typo to me. Both of the
megasas_read_fw_status_reg_* functions involved are identical though, so
there was no bad behaviour. I changed it for consistency and clarity.

Signed-off-by: James Georgas <soulpa7ch@gmail.com>
Acked-by: Sumit Saxena <sumit.saxena@lsi.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
(cherry picked from commit a3fda7dd5179989dd0ead820dcebd13f956ddec1)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/megaraid/megaraid_sas_base.c b/drivers/scsi/megaraid/megaraid_sas_base.c
index 8dff124..e1eeaad 100644
--- a/drivers/scsi/megaraid/megaraid_sas_base.c
+++ b/drivers/scsi/megaraid/megaraid_sas_base.c
@@ -596,7 +596,7 @@ megasas_clear_intr_skinny(struct megasas_register_set __iomem *regs)
  /*
   * Check if it is our interrupt
   */
- if ((megasas_read_fw_status_reg_gen2(regs) & MFI_STATE_MASK) ==
+ if ((megasas_read_fw_status_reg_skinny(regs) & MFI_STATE_MASK) ==
      MFI_STATE_FAULT) {
   mfiStatus = MFI_INTR_FLAG_FIRMWARE_STATE_CHANGE;
  } else
-- 
1.7.1